video_xblock/static/vendor/js/Vimeo.js

```diff
@@ -156,28 +156,36 @@ THE SOFTWARE. */
       this.trigger('progress');
       if (durationUpdate) this.trigger('durationchange');
     },
+
     onPlayProgress: function(data) {
       this.vimeoInfo.time = data.seconds;
       this.trigger('timeupdate');
     },
+
     onPlay: function() {
       this.vimeoInfo.state = VimeoState.PLAYING;
       this.trigger('play');
     },
+
     onPause: function() {
       this.vimeoInfo.state = VimeoState.PAUSED;
       this.trigger('pause');
     },
+
     onFinish: function() {
       this.vimeoInfo.state = VimeoState.ENDED;
       this.trigger('ended');
     },
+
     onSeek: function(data) {
       this.trigger('seeking');
       this.vimeoInfo.time = data.seconds;
+      this.vimeoInfo.duration = data.duration;
+      this.trigger("durationchange");
```

Comment:
I see you find a workaround for the headache. Nice if it works well. But it looks like as a workaround🙂

Reply:
In this case, I use callback function from videoJS library which we use for render video player